File: java_folder/assignment1.java
package java_folder;


import java.util.*;

public class assignment1 {
    private static double balance=1000.00;
    private static int pin =1234;
    private static ArrayList<String> transactionHistory = new ArrayList<>();
    private static void balanceInquiry() 
    {
        System.out.println("Current balance: $" + balance);
        transactionHistory.add("Balance Inquiry: $" + balance);
    }

    private static void cashWithdrawal(Scanner sc) 
    {
        System.out.print("Enter withdrawal amount: $");
        double amount = sc.nextDouble();

        if (amount > balance) {
            System.out.println("Insufficient funds.");
        } else {
            balance -= amount;
            System.out.println("Withdrawal successful. New balance: $" + balance);
            transactionHistory.add("Withdrawal: $" + amount);
        }
    }

    private static void cashDeposit(Scanner sc) 
    {
        System.out.print("Enter deposit amount: $");
        double amount = sc.nextDouble();

        balance += amount;
        System.out.println("Deposit successful. New balance: $" + balance);
        transactionHistory.add("Deposit: $" + amount);
    }

    private static void changePIN(Scanner sc)
    {
        System.out.print("Enter new PIN: ");
        int newPIN = sc.nextInt();

        pin = newPIN;
        System.out.println("PIN successfully changed.");
        transactionHistory.add("PIN Change");
    }

    private static void displayTransactionHistory()
    {
        System.out.println("Transaction History:");
        for (String transaction : transactionHistory) {
            System.out.println(transaction);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter your PIN: ");
        int enterpin=sc.nextInt();
        if(enterpin==pin)
        {
            int option =0;
            while(option!=6)
            {
                System.out.println("\nATM Menu:");
                System.out.println("1. Balance Inquiry");
                System.out.println("2. Cash Withdrawal");
                System.out.println("3. Cash Deposit");
                System.out.println("4. Change PIN");
                System.out.println("5. Transaction History");
                System.out.println("6. Exit");

                System.out.print("Choose an option: ");
                option = sc.nextInt();
                switch (option) {
                    case 1:
                        balanceInquiry();
                        break;
                    case 2:
                        cashWithdrawal(sc);
                        break;
                    case 3:
                        cashDeposit(sc);
                        break;
                    case 4:
                        changePIN(sc);
                        break;
                    case 5:
                        displayTransactionHistory();
                        break;
                    case 6:
                        System.out.println("Thank you for using the ATM. Goodbye!");
                        break;
                    default:
                        System.out.println("Invalid option. Please try again.");
                }
            }
        }
        else
        {
            System.out.println("Incorrect PIN. Please try again.");
        }
        sc.close();
    }
}

File: python/Assignment1.py
class ATMMACHINE:
    def __init__(self):
        self.balance = 1000.00
        self.pin = "1234"
        self.transaction_history = []
    
    def run(self):
        Enterpin = input("ENTER YOUR PIN: ")
        if Enterpin == self.pin:
            while True:
                print("\nATM Menu:")
                print("1. Check Balance")
                print("2. Withdraw Money")
                print("3. Deposit Money")
                print("4. Change PIN")
                print("5. View Transaction History")
                print("6. Exit")

                choice = input("ENTER YOUR CHOICE: ")

                if choice == '1':
                    self.check_balance()
                elif choice == '2':
                    self.withdraw_money()
                elif choice == '3':
                    self.money_deposit()
                elif choice == '4':
                    self.change_pin()
                elif choice == '5':
                    self.view_transaction_history()
                elif choice == '6':
                    print("Thank you for using the ATM. Goodbye!")
                    break
                else:
                    print("Invalid option. Please try again.")
        else:
            print("Incorrect PIN.")
    
    def check_balance(self):
        print(f"Your balance is: ${self.balance:.2f}")
        self.transaction_history.append(f"Checked balance: ${self.balance:.2f}")

    def withdraw_money(self):
        amount = float(input("Enter amount to withdraw: $"))
        if amount > self.balance:
            print("Insufficient balance")
        else:
            self.balance -= amount
            print(f"Withdrawal successful. New balance: ${self.balance:.2f}")
            self.transaction_history.append(f"Withdrew: ${amount:.2f}")

    def money_deposit(self):
        amount = float(input("Enter amount to deposit: $"))
        self.balance += amount
        print(f"Deposit successful. New balance: ${self.balance:.2f}")
        self.transaction_history.append(f"Deposited: ${amount:.2f}")

    def change_pin(self):
        new_pin = input("Enter new PIN: ")
        self.pin = new_pin
        print("PIN changed successfully.")
        self.transaction_history.append("PIN changed")

    def view_transaction_history(self):
        print("\nTransaction History:")
        if not self.transaction_history:
            print("No transactions available.")
        else:
            for transaction in self.transaction_history:
                print(transaction)


if __name__ == "__main__":
    atm = ATMMACHINE()
    atm.run()
